src/dataflow/util/DataflowUtils.java

```diff
@@ -25,29 +28,29 @@
  */
 public class DataflowUtils {
 
-    public static String[] getTypeNames(AnnotationMirror type) {
-        return getDataflowValue(type, "typeNames");
+    /** The DataFlow.typeNames element/field. */
+    private final ExecutableElement dataflowTypeNamesElement;
+
+    /** The DataFlow.typeNameRoots element/field. */
+    private final ExecutableElement dataflowTypeNameRootsElement;
+
+    public DataflowUtils(ProcessingEnvironment processingEnv) {
+        dataflowTypeNamesElement = TreeUtils.getMethod(DataFlow.class, "typeNames", processingEnv);
+        dataflowTypeNameRootsElement = TreeUtils.getMethod(DataFlow.class, "typeNameRoots", processingEnv);
     }
 
-    public static String[] getTypeNameRoots(AnnotationMirror type) {
-        return getDataflowValue(type, "typeNameRoots");
+    public List<String> getTypeNames(AnnotationMirror type) {
+        return AnnotationUtils.getElementValueArray(
+                type, dataflowTypeNamesElement, String.class, Collections.emptyList());
     }
 
-    private static String[] getDataflowValue(AnnotationMirror type, String valueName) {
-        List<String> allTypesList = AnnotationUtils.getElementValueArray(type, valueName, String.class,
-                true);
-        // types in this list is org.checkerframework.framework.util.AnnotationBuilder.
-        String[] allTypesInArray = new String[allTypesList.size()];
-        int i = 0;
-        for (Object o : allTypesList) {
-            allTypesInArray[i] = o.toString();
-            i++;
-        }
-        return allTypesInArray;
+    public List<String> getTypeNameRoots(AnnotationMirror type) {
+        return AnnotationUtils.getElementValueArray(
+                type, dataflowTypeNameRootsElement, String.class, Collections.emptyList());
     }
 
     public static AnnotationMirror createDataflowAnnotationForByte(String[] dataType,
-            ProcessingEnvironment processingEnv) {
+                                                                   ProcessingEnvironment processingEnv) {
```

Comment:
Maybe we don't force reformat as a pre-commit hook? Can you see what happens with this if you reformat?

Reply:
We'll need to apply AOSP code style in future.